encoder.go

```diff
@@ -25,13 +25,13 @@ import (
 	"time"
 )
 
-// encoder is a format-agnostic interface for all log field encoders. It's not
+// Encoder is a format-agnostic interface for all log field encoders. It's not
 // safe for concurrent use.
-type encoder interface {
+type Encoder interface {
```

Comment:
If we are exporting this, let's add comments on the below methods, especially `Clone`/`Free` since we're expecting the implementors to do the pooling.